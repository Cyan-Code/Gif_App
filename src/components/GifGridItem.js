import React from 'react'

export const GifGridItem = ( img ) => {

  return (
    <div className="card animate__fadeIn">
      <p> {img.title} </p>
      <img src={img.url} alt={img.title} />
    </div>
  )
}