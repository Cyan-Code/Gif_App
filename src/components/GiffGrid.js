import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GiffGrid = ({ category }) => {

  const {data, loading} = useFetchGifs(category)

  return (
    <>
      <h3 className="card animate__fadeIn"> {category} </h3>
      
      { loading && <p className = "card animate__flash">Loading</p>}

      <div className="card-grid">
        <ol>
          <li>
            {
              data.map ( img => (
                <GifGridItem
                  key = { img.id }
                  { ...img }
                />
              ))
            }
          </li>
        </ol>
      </div>
    </>
  )
}

GiffGrid.propTypes = {
  category: PropTypes.string.isRequired
}
