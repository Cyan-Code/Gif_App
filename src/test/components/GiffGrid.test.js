import React from 'react';
import { GiffGrid } from '../../components/GiffGrid';
import {shallow} from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre <GiffGrid />', () => {
  
  const category = 'hola'

  test('Debe renderizar el componente de manera correcta', () => {

  })
  
  //Pruebas sobre useFetchGifs
  test('Debe de mostrar items cuando se carga imÃ¡genes de useFetch', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/algo.png',
      title: 'cualeuri cosa'
    },
    {
      id: 'A',
      url: 'https://localhost/algo.png',
      title: 'cualeuri cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: true
    })

    const wrapper = shallow (<GiffGrid category={ category }/>)

    expect( wrapper.find('p').exists() ).toBe(true)
    expect( wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
  

})

