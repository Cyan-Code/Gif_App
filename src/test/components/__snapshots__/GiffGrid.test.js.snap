// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GiffGrid /> Debe de mostrar items cuando se carga im√°genes de useFetch 1`] = `
<Fragment>
  <h3
    className="card animate__fadeIn"
  >
     
    hola
     
  </h3>
  <p
    className="card animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol>
      <li>
        <GifGridItem
          id="ABC"
          key="ABC"
          title="cualeuri cosa"
          url="https://localhost/algo.png"
        />
      </li>
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas sobre <GiffGrid /> Debe renderizar el componente de manera correcta 1`] = `
<Fragment>
  <h3
    className="card animate__fadeIn"
  >
     
    hola
     
  </h3>
  <p
    className="card animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol>
      <li />
    </ol>
  </div>
</Fragment>
`;
